src/modules/hip/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,164 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ __forceinline__ float calcMel(float *input, int melBin,
+                                         float *weightsDown, int *intervals,
+                                         int fftStride, int fftShift,
+                                         float normFactor)
+{
+    float out = 0;
+    int fftbin = intervals[melBin];
+    int fftBinEnd = intervals[melBin + 1];
+
+    float *in =  input + fftbin * fftStride + fftShift;
+    for (; fftbin < fftBinEnd; fftbin++, in += fftStride) 
+    {
+        auto weightUp = float(1) - weightsDown[fftbin];
+        weightUp *= normFactor;
+        out += *in * weightUp;
+    }
+
+    fftBinEnd = intervals[melBin + 2];
+    in =  input + fftbin * fftStride + fftShift;
+
+    for (; fftbin < fftBinEnd; ++fftbin, in += fftStride) 
+    {
+        auto weightDown = weightsDown[fftbin];
+        weightDown *= normFactor;
+        out += *in * weightDown;
+    }
+
+    return out;
```

Comment:
Please [ass output as a argument to this function and write directly inside the function, instead of returning the value

Reply:
Done